src/mpi/init/finalize.c

```diff
@@ -331,14 +331,6 @@ int MPI_Finalize(void)
     }
 #endif
 
-    /* Finalize the threading library after releasing all synchronization
```

Comment:
`MPL_thread_finalize` (directly) finalizes Argobots, so it must be called in `MPI_Finalize`.
Can we put `MPICH_IS_THREADED` here?